.github/workflows/nextjs.yml

```
@@ -0,0 +1,37 @@
+name: Next.js CI
+
+on:
+  pull_request:
+    branches: [main]
+
+jobs:
+  next:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+          cache: "yarn"
+
+      - name: Install dependencies
+        run: yarn --frozen-lockfile
```

It's better to use `yarn --immutable` for forward-compatibility, [as `yarn --frozen-lockfile` will possibly be removed later in the future versions of Yarn](https://yarnpkg.com/cli/install).

```suggestion
        run: yarn --immutable
```